### ISSUE TITLE
ConnectionClosed should exist as a superclass of ConnectionDone and ConnectionLost

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3137]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)| @radix reported|
|-|-|
|Trac ID|trac#3137|
|Type|enhancement|
|Created|2008-03-29 17:57:36Z|
|Branch|https://github.com/twisted/twisted/tree/connection-closed-3137|

I'm not sure if this is really necessary, but it would be handy for gTwisted, where we see tons of "except (ConnectionLost, ConnectionDone):".

<details><summary>Searchable metadata</summary>

```
trac-id__3137 3137
type__enhancement enhancement
reporter__radix radix
priority__normal normal
milestone__ 
branch__branches_connection_closed_3137 branches/connection-closed-3137
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1206813456000000 1206813456000000
changetime__1206891294000000 1206891294000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_1"></a>|@radix commented|
|-|-|

(In [23123]) Branching to 'connection-closed-3137'

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_2"></a>|@radix removed owner|
|-|-|

Ready for review.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_3"></a>|@therve set owner to @radix|
|-|-|

 * please update the copyright of both files
 * you can remove the unittest.main() at the end of test_error

Then please merge.

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_4"></a>|@radix set status to `closed`|
|-|-|

(In [23138]) Merge connection-closed-3137

Author: radix
Reviewer: therve
Fixes [#3137](https://github.com/twisted/twisted/issues/3137)

ConnectionClosed is a new exception which is a superclass of ConnectionLost
and ConnectionDone. This makes it easier to write code which wants to handle
both errors in the same way.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
